### ISSUE TITLE
What to do in case of detecting a problem with a file

### ISSUE THREAD

**Issue by guille2306** [guille2306/backuppc_check_pool#3]
Question received by email:

> We have one question : what to do if/when the script detects a problem
on a pooled file ? Can we just delete the file (based on the -maybe
wrong- assumption it will be copied back on the next run) ?
>
> Tia,
> B&A Consultants - Sécurité informatique - Conseils et audits
> www.ba-consultants.fr

**Comment by guille2306** [comment]
Good question. Sincerely, I don't know. I've pushed back that question to the moment I actually find a problem, and I hope that moment never comes ;)

I don't think deleting the file would make any difference, for the same reason a corrupted file in the pool is not picked-up by BackupPC. Probably the best would be to "touch" in some way the original file (in the client) and wait for BackupPC to copy it again. This won't solve the corrupted file for previous backups, though.

One extra point: given the MD5 checksum (the name of the pool file), it's not trivial to find the original location and name of the file: https://github.com/backuppc/backuppc/wiki/How-to-find-which-backups-reference-a-particular-pool-file.

**Comment by SpareSimian** [comment]
Perhaps capturing the find logic in another script would be a useful endeavor. It would loop over all hosts/shares/generations to locate the matching checksum. (What happens when there's an (admittedly-rare) checksum collision?)

**Comment by guille2306** [comment]
Yes, that's the basic process described in the BackupPC link above. First check which hosts and backups reference the MD5 (relatively fast and easy), and then loop over them and search the entire backup tree for that digest (relatively easy but very slow).
